lib/src/incrementally_loading_listview.dart

```diff
@@ -68,6 +68,44 @@ class IncrementallyLoadingListView extends StatefulWidget {
       this.onLoadMore,
       this.onLoadMoreFinished});
 
+  IncrementallyLoadingListView.separated(
+      {@required this.hasMore,
+      @required this.loadMore,
+      this.loadMoreOffsetFromBottom = 0,
+      this.key,
+      this.scrollDirection = Axis.vertical,
+      this.reverse = false,
+      this.controller,
+      this.primary,
+      this.physics,
+      this.shrinkWrap = false,
+      this.padding,
+      this.itemExtent,
+      @required IndexedWidgetBuilder itemBuilder,
+      @required IndexedWidgetBuilder separatorBuilder,
+      @required ItemCount itemCount,
+      this.addAutomaticKeepAlives = true,
+      this.addRepaintBoundaries = true,
+      this.cacheExtent,
+      this.onLoadMore,
+      this.onLoadMoreFinished})
+      : this.itemCount = (() {
+          final count = itemCount();
```

Comment:
Sorry noticed something else where what the user is passing for `itemCount` is getting changed. I get why it is done but this is a public property and is something the user specified so should be kept as is. Same thing for `itemBuilder` That the number of items is being increased is something that is better handled internally. By that mean, to see if all of this logic could be moved to be around where the widget gets built and calls `ListView.builder(...)`